### PR TITLE
Option to enable/disable Jolokia agent to collect JMX stat

### PR DESCRIPTION
**Pull request by bijugs**
Since ``Jolokia`` agent is used to collect JMX stats from ``kafka``, want to use the same for the ``ZooKeeper`` processes on the ``Kafka`` clusters. By default enabling ``Jolokia`` agent is set to ``false`` so that ``jmx-agents`` will be used. When enabled, ``jmx-agents`` need to be [disabled](https://github.com/bloomberg/chef-bach/blob/56c6c208629e34c9a778bad75e489a360bcbcaaf/cookbooks/bcpc-hadoop/attributes/default.rb#L38).

*Note:* Not tested yet. Raised this to get early feedback.

### CODE REVIEW

**Comment by http-418** [comment]
Looks good to me.

File: cookbooks/bcpc-hadoop/recipes/zookeeper_impl.rb [hunk 1]
```diff
@@ -25,6 +25,11 @@
   to "/usr/hdp/current/zookeeper-client/bin/zookeeper-server-initialize"
 end
 
+#Install jolokia's jvm agent to node['bcpc']['jolokia']['path']
+if node[:bcpc][:jolokia][:enable] == true
+  include_recipe 'bcpc-hadoop::jolokia'
```
Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/recipes/zookeeper_impl.rb]: This looks duplicated with [bcpc_kafka::zookeeper_server](https://github.com/bloomberg/chef-bach/blob/8952764c95cf77063f2303a0b08231a4599fd1a3/cookbooks/bcpc_kafka/recipes/zookeeper_server.rb#L18)?

Comment (bijugs) [reply cookbooks/bcpc-hadoop/recipes/zookeeper_impl.rb]: Thanks @cbaenziger . Updated [bcpc_kafka::zookeeper_server](https://github.com/bloomberg/chef-bach/blob/8952764c95cf77063f2303a0b08231a4599fd1a3/cookbooks/bcpc_kafka/recipes/zookeeper_server.rb#L17)

**Comment by cbaenziger** [comment]
Thanks Biju; please update once tested.

**Comment by vt0r** [comment]
Do we need to use `force_override` in the 3-ish places I think I saw it being used? Shouldn't `force_default` suffice?

**Comment by bijugs** [comment]
Updated PR to download ``Jolokia`` agent jar instead of using ``maven``.

**Comment by http-418** [comment]
I would rather not add anything new to the "bins" tarball.  In a perfect world, the "bins" tarball would not exist.  It is one of the major obstacles to getting rid of "bootstrap" hosts and using shared infrastructure.

Besides, Jolokia is itself distributed as a jar file via maven central.  The maven cookbook is the natural and obvious way to distribute it to our hosts.  Why would we go out of our way to re-host this and do it in an awkward way?

**Comment by bijugs** [comment]
@http-418 , as discussed offline, the code to re-host was done due to concerns raised about not being able to build VM clusters being a proxy which seem to be the cause of [this](https://github.com/bloomberg/chef-bach/commit/8952764c95cf77063f2303a0b08231a4599fd1a3#diff-ab0bd0575946adf51413dd40f07f4324L28) removal earlier. Once we are able to build a VM cluster behind the proxy with out re-hosting Jolokia jar, will remove the last commit from the PR.

**Comment by http-418** [comment]
It appears the hdfsdu PR added a lot of the necessary SSL stuff for maven:
https://github.com/bloomberg/chef-bach/blob/master/cookbooks/bcpc-hadoop/recipes/maven_config.rb

I'm still not sure how it works -- are SSL certs downloaded from the bootstrap?

**Comment by cbaenziger** [comment]
The code will setup Maven using the SSL certs if available. (E.g. as we setup on the bootstrap.) But today, no Maven commands are run on the cluster nodes, so no infrastructure yet exists to ship the certs from bootstrap to cluster nodes for them to run Maven.

**Comment by vt0r** [comment]
Closing in favor of #1149

**Pull request closed without merging**